magma/bits.py

```diff
@@ -66,6 +66,17 @@ def __str__(cls):
 class Bits(Array, AbstractBitVector, metaclass=BitsMeta):
     __hash__ = Array.__hash__
 
+    def __init__(self, *args, **kwargs):
+        if args and len(args) == 1 and isinstance(args[0], m.Array) and \
+                len(self) > 1 and len(args[0]) <= len(self):
+            self.ts = args[0].ts[:]
+            # zext for promoting width
```

Comment:
py/smt also use zext but I think this might not be the most obvious behavior.  We should ext so that signed values are extended correctly. Or perhaps we should use the ext of the current type? not sure.

Reply:
That makes sense, I can update the SInt constructor sext for now.  We can always remove this code if we decide to not support the behavior, but it seems reasonable for now and the lassen code uses that behavior